app/models/reservation.rb

```diff
@@ -87,10 +87,16 @@ def status # rubocop:disable CyclomaticComplexity, PerceivedComplexity
     end
   end
 
+  delegate :category_id, to: :equipment_model
+
   def duration
     due_date.to_date - start_date.to_date + 1
   end
 
+  def time_checked_out
+    checked_in.to_date - checked_out.to_date + 1 if checked_out && checked_in
```

Comment:
Does the fact that this returns `nil` for either future, missed, or checked out reservations give us any problems anywhere?